builtins/web/form-data/form-data-encoder.cpp

```diff
@@ -42,6 +43,33 @@ size_t compute_normalized_len(std::string_view src) {
   return len;
 }
 
+std::optional<size_t> compute_unencoded_normalized_len(JSContext *cx, JS::HandleString value) {
```

Comment:
I ran into an issue resolving with the `mozilla::Result` type at this level and instead am using `std::nullopt` as an OOM indicator. If that isn't the right choice I can move this function into the namespace where OutOfMemory is defined.